sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -657,6 +659,18 @@ class Analyzer(
     }
   }
 
+  /**
+   * Resolve table relations with concrete relations from v2 catalog.
```

Comment:
can we add a little more notes about how this rule interacts with v1 table resolution rules?

Reply:
Will do